### PR TITLE
Add a default config file

### LINKED ISSUE #226: Add a default config file which put most things into comment with description.
@yanxuean

### PR DESCRIPTION
**Pull request by yanxuean**
fix  #226 
Signed-off-by: yanxuean <yan.xuean@zte.com.cn>

@Random-Liu

### CODE REVIEW

**Comment by yanxuean** [comment]
test record
# _output/cri-containerd --default-config
socket_path = "/var/run/cri-containerd.sock"
root_dir = "/var/lib/cri-containerd"
stream_server_address = ""
stream_server_port = "10010"
cgroup_path = ""
enable_selinux = false
sandbox_image = "gcr.io/google_containers/pause:3.0"

[containerd]
  snapshotter = "overlayfs"
  endpoint = "/run/containerd/containerd.sock"

[cni]
  bin_dir = "/opt/cni/bin"
  conf_dir = "/etc/cni/net.d"

**Comment by Random-Liu** [comment]
@yanxuean This is nice! Could you rebase this PR? The previous one is merged. :)

**Comment by yanxuean** [comment]
@Random-Liu 
done

File: cmd/cri-containerd/cri_containerd.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ func main() {
 		os.Exit(0)
 	}
 
+	if o.PrintDefaultConfig {
+		o.GetDefaultTomlConfig()
```
Comment (Random-Liu) [inline cmd/cri-containerd/cri_containerd.go]: Let's call it `PrintDefaultTomlConfig`.

File: cmd/cri-containerd/options/options.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ import (
 	"flag"
 	"os"
 
+	"fmt"
```
Comment (Random-Liu) [inline cmd/cri-containerd/options/options.go]: Move up.

File: cmd/cri-containerd/options/options.go [hunk 3]
```diff
@@ -109,6 +112,8 @@ func (c *CRIContainerdOptions) AddFlags(fs *pflag.FlagSet) {
 		false, "Enable selinux support.")
 	fs.StringVar(&c.SandboxImage, "sandbox-image",
 		"gcr.io/google_containers/pause:3.0", "The image used by sandbox container.")
+	fs.BoolVar(&c.PrintDefaultConfig, "default-config",
+		false, "print default toml config of cri-containerd and quit.")
```
Comment (Random-Liu) [inline cmd/cri-containerd/options/options.go]: s/print/Print

File: cmd/cri-containerd/options/options.go [hunk 4]
```diff
@@ -145,3 +150,22 @@ func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
...
 	return fs.Parse(commandline)
 }
+
+// GetDefaultTomlConfig print default toml config of cri-containerd.
+func (c *CRIContainerdOptions) GetDefaultTomlConfig() {
+	fs := pflag.NewFlagSet(os.Args[0]+"_getDefaultConfig", pflag.ExitOnError)
+
+	c.AddFlags(fs)
+
+	err := fs.Parse([]string{})
```
Comment (Random-Liu) [inline cmd/cri-containerd/options/options.go]: nit:
```
if err := fs.Parse([]string{}); err != nil {
}
```
To save one line. And also for the following one.

**Comment by Random-Liu** [comment]
@yanxuean Can we have a default config object? And get the default from the default config object in `AddFlags`.

And then `PrintDefaultConfig` could just encode the default config object and print directly.

**Comment by yanxuean** [comment]
Actualy there are no need to do Parse,  it has set the default value to variable in AddFlags by pflag source code.But it belong to pflag's internal implement.Pflag do not promise it for user. So I think we should not leverage it. And you?

File: cmd/cri-containerd/cri_containerd.go (hunk 1 above)
Comment (yanxuean) [reply cmd/cri-containerd/cri_containerd.go]: done

File: cmd/cri-containerd/options/options.go (hunk 2 above)
Comment (yanxuean) [reply cmd/cri-containerd/options/options.go]: done

File: cmd/cri-containerd/options/options.go (hunk 3 above)
Comment (yanxuean) [reply cmd/cri-containerd/options/options.go]: done

File: cmd/cri-containerd/options/options.go (hunk 4 above)
Comment (yanxuean) [reply cmd/cri-containerd/options/options.go]: done

File: cmd/cri-containerd/options/options.go [hunk 5]
```diff
@@ -145,3 +148,20 @@ func (c *CRIContainerdOptions) InitFlags(fs *pflag.FlagSet) error {
 	// So I trigger twice parses, one for default value, one for commandline value.
 	return fs.Parse(commandline)
 }
+
+// PrintDefaultTomlConfig print default toml config of cri-containerd.
+func (c *CRIContainerdOptions) PrintDefaultTomlConfig() {
+	fs := pflag.NewFlagSet(os.Args[0]+"_getDefaultConfig", pflag.ExitOnError)
```
Comment (Random-Liu) [inline cmd/cri-containerd/options/options.go]: nit: Just name it "default-config" if fine I feel like, we never use it anyway.

**Comment by Random-Liu** [comment]
> Actualy there are no need to do Parse, it has set the default value to variable in AddFlags by pflag source code.But it belong to pflag's internal implement.Pflag do not promise it for user. So I think we should not leverage it. And you?

Golang is doing the same thing. I think it's fine to remove the `Parse`.

**Comment by Random-Liu** [comment]
2 nits, otherwise LGTM.

**Comment by Random-Liu** [comment]
LGTM

**Pull request merged**